src/System.Management.Automation/engine/Utils.cs

```diff
@@ -650,30 +650,31 @@ internal static string GetRegistryConfigurationPath(string shellID)
 #else
             lock (s_cachedGroupPolicySettings)
             {
-                // Return cached information, if we have it
-                Dictionary<string, object> settings;
-                if ((s_cachedGroupPolicySettings.TryGetValue(settingName, out settings)) &&
```

Comment:
If we don't find the cached item the first time through the loop, but do the second iteration, your way will:
* copy the setting name twice
* concat the searchKey twice
* hash that value twice
* read from the registry once

In contrast, the tuple will:
* allocate a single tuple (as opposed to two strings)
* combine the hash of 3 fields (combine is cheap, a shift, add, and xor per field)
* avoid reading the registry

Significantly faster? Who knows - but it seems better (does less) and I think it's simpler code.